src/components/figures/chart/chartJs/chartjs_show_values_plugin.ts

```diff
@@ -29,74 +30,142 @@ export const chartShowValuesPlugin: Plugin = {
+          yPosition = otherPosition - 13;
         }
-        case "bar":
-        case "line": {
-          const yOffset = dataset.type === "bar" && !options.horizontal ? 0 : 3;
-          for (let i = 0; i < dataset._parsed.length; i++) {
-            const point = dataset.data[i];
-            const value = options.horizontal ? dataset._parsed[i].x : dataset._parsed[i].y;
-            const displayedValue = options.callback(value - 0);
-            let xPosition = 0,
-              yPosition = 0;
-            if (options.horizontal) {
-              yPosition = point.y;
-              if (value < 0) {
-                ctx.textAlign = "right";
-                xPosition = point.x - yOffset;
-              } else {
-                ctx.textAlign = "left";
-                xPosition = point.x + yOffset;
-              }
-            } else {
-              xPosition = point.x;
-              if (value < 0) {
-                ctx.textBaseline = "top";
-                yPosition = point.y + yOffset;
-              } else {
-                ctx.textBaseline = "bottom";
-                yPosition = point.y - yOffset;
-              }
-            }
-            ctx.strokeText(displayedValue, xPosition, yPosition);
-            ctx.fillText(displayedValue, xPosition, yPosition);
-          }
-          break;
+      }
+      textsPositions[xPosition].push(yPosition);
+
+      ctx.fillStyle = point.options.backgroundColor;
+      ctx.strokeStyle = options.background || "#ffffff";
+      drawTextWithBackground(options.callback(value - 0), xPosition, yPosition, ctx);
+    }
+  }
+}
+
+function drawHorizontalBarChartValues(
+  chart: any,
+  options: ChartShowValuesPluginOptions,
+  ctx: CanvasRenderingContext2D
+) {
+  const xMax = chart.chartArea.right;
+  const xMin = chart.chartArea.left;
+  const textsPositions: Record<number, number[]> = {};
+
+  for (const dataset of chart._metasets) {
+    for (let i = 0; i < dataset._parsed.length; i++) {
+      const value = dataset._parsed[i].x;
+      const displayValue = options.callback(value - 0);
+      const point = dataset.data[i];
+
+      const yPosition = point.y;
+      let xPosition = value < 0 ? point.x + point.width / 2 : point.x - point.width / 2;
+      xPosition = Math.min(xPosition, xMax);
+      xPosition = Math.max(xPosition, xMin);
+
+      // Avoid overlapping texts with same Y
+      if (!textsPositions[yPosition]) {
+        textsPositions[yPosition] = [];
+      }
+      const textWidth = computeTextWidth(ctx, displayValue, { fontSize: 12 }, "px");
+      for (const otherPosition of textsPositions[yPosition]) {
+        if (Math.abs(otherPosition - xPosition) < textWidth) {
+          xPosition = otherPosition + textWidth + 3;
         }
       }
-    });
+      textsPositions[yPosition].push(xPosition);
 
-    ctx.restore();
-  },
-};
+      ctx.fillStyle = point.options.backgroundColor;
+      ctx.strokeStyle = options.background || "#ffffff";
```

Comment:
Not 100% sure about this one, if you have a very light grey dataset color and no background, the text value becomes unreadable. Maybe having the stroke computed based on `point.options.backgroundColor` would be better to ensure the text is visible?
fyi, there are 2 datasets in the picture :p
![image](https://github.com/user-attachments/assets/c65c12c9-3202-4401-9353-5a90ef57132b)

Reply:
Having the text stroke be the same color as the background is _ususally_ prettier. Here for example if we do a white or black text bacgrkound it's not as pretty : 
![image](https://github.com/user-attachments/assets/d7beabf9-39e8-4e3a-9af3-34da8fe40138)
![image](https://github.com/user-attachments/assets/ccfd0cd6-eb33-41ea-827e-e67364805a7d)

But indeed there are cases where it'll be ugly.  I was thinking that if the user pick shitty colors that's his problem, GSheet does the same: 
![image](https://github.com/user-attachments/assets/bfaa9a4c-e0d1-4702-8634-f08347539d60)

I'll ask Francois

Reply:
![image](https://github.com/user-attachments/assets/6c0b25e6-edeb-443d-b621-7a3c9dd0edb4)